examples/src/main/scala/org/apache/spark/examples/ml/DeveloperApiExample.scala

```diff
@@ -176,9 +172,7 @@ private class MyLogisticRegressionModel(
    *
    * This is used for the default implementation of [[transform()]].
    */
-  override protected def copy(): MyLogisticRegressionModel = {
-    val m = new MyLogisticRegressionModel(parent, fittingParamMap, weights)
-    Params.inheritValues(extractParamMap(), this, m)
-    m
+  override def copy(extra: ParamMap): MyLogisticRegressionModel = {
+    copyValues(new MyLogisticRegressionModel(parent, weights), extra)
   }
```

Comment:
Do you think it will be more consistent to most of Java/Scala APIs by asking users to implement `clone` method, and we just have default `copy` method?
```scala
def copy(extra: ParamMap): MyLogisticRegressionModel = {
  copyValues(this.clone(), extra)		
}
```